File: Java/app/src/main/java/com/example/simpleasynctask/MainActivity.java
package com.example.simpleasynctask;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.ProgressBar;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    // Key for saving the state of the TextView
    private static final String TEXT_STATE = "currentText";

    // The TextView where we will show results
    private TextView mTextView;
    private ProgressBar progressBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mTextView = findViewById(R.id.textView1);
        progressBar = findViewById(R.id.progressBar);

        // Restore TextView if there is a savedInstanceState bundle.
        if (savedInstanceState != null) {
            mTextView.setText(savedInstanceState.getString(TEXT_STATE));
        }
    }

    public void beginTask(View view) {
        // Put a message in the text view.
        mTextView.setText(R.string.napping);

        progressBar.setMax(10);
        progressBar.setProgress(0);

        // Start the AsyncTask.
        // The AsyncTask has a callback that will update the text view.
        new SimpleAsyncTask(mTextView, progressBar).execute(10);
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        // Save the state of the TextView
        outState.putString(TEXT_STATE, mTextView.getText().toString());
    }
}

File: Kotlin/app/src/main/java/com/example/asynctask/MainActivity.kt
package com.example.asynctask

import android.os.Bundle
import android.view.View
import android.widget.ProgressBar
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity() : AppCompatActivity(){
    private var mTextView: TextView? = null
    private var TEXT_STATE = "currentText"
    private var progressBar: ProgressBar? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mTextView = findViewById(R.id.textView1)
        progressBar = findViewById(R.id.progressBar)

        with(mTextView) { this?.setText(savedInstanceState?.getString(TEXT_STATE)) }
    }

    fun startTask(view: View?) { // Put a message in the text view
        mTextView?.setText(R.string.napping)
        progressBar?.setMax(10)
        progressBar?.setProgress(0)

        // Start the AsyncTask.
        val task = SimpleAsyncTask()
        task.execute(10)
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        // Save the state of the TextView
        outState.putString(TEXT_STATE, mTextView!!.text.toString())
    }
}
